### DISCUSSION TITLE
How do I make multiple mermaid diagrams as subfigures?

### DISCUSSION THREAD

**Discussion by roaldarbol** [quarto-dev/quarto-cli#11482] [Q&A] [answered]
### Description

I am trying to create a single figure that contains multiple mermaid diagrams as subfigures (as in https://quarto.org/docs/authoring/figures.html#subfigures). I can get both to render, but not as subfigures. Here's a code snippet:

````
:::{#fig-intro-diagram layout-ncol=2}

```{mermaid}
flowchart TD
  A[High-level behaviour] --> B[Low-level behaviour]
```

```{mermaid}
flowchart TD
  A[High-level behaviour] --> B[Low-level behaviour]
  C[Environment] -.-> A & B
```

This is a caption.
:::
````

Is there a way of achieving that?

**Comment by roaldarbol** [comment, marked as answer]
Ah, figured a way!
Since subfigures need their own labels/tags, and the way to achieve that is through a figure div, you can simply nest figure divs:

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Subfigures"
format: html
---

::::{#fig-intro-diagram layout-ncol=2}

:::{#fig-intro-1}
```{mermaid}
flowchart TD
  A[High-level behaviour] --> B[Low-level behaviour]
```
:::

:::{#fig-intro-2}
```{mermaid}
flowchart TD
  A[High-level behaviour] --> B[Low-level behaviour]
  C[Environment] -.-> A & B
```
:::

This is a caption.
::::

````

</td><td>

<img width="662" alt="image" src="https://github.com/user-attachments/assets/81c63cb2-d5ac-4340-8219-801ae76eb42a">

</td></tr>
</table>

**Comment by mcanouil** [comment]
Don't mix components, nest them.

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format: html
---

::: {#fig-intro-diagram}

::: {layout-ncol=2}

```{mermaid}
flowchart TD
  A[High-level behaviour] --> B[Low-level behaviour]
```

```{mermaid}
flowchart TD
  A[High-level behaviour] --> B[Low-level behaviour]
  C[Environment] -.-> A & B
```
:::

This is a caption.
:::

````

</td><td>

<img width="662" alt="image" src="https://github.com/user-attachments/assets/ac4650ae-786f-4123-99f4-c57ddaafcbd1">

</td></tr>
</table>

**Reply by roaldarbol** [reply]
The difference is that I don't get `(a)` and `(b)` subfigures that I can refer to this way.

**Reply by cscheid** [reply]
That's right. I think your approach is slightly better in this case, @roaldarbol.

**Reply by mcanouil** [reply]
It depends on the goal. To allow cross-reference of subfigures and have subcaptions, using cross-reference div is the way. If one wants a composite figure then the nested syntax is the way.

**Reply by cscheid** [reply]
(@mcanouil, the title says "subfigures")

**Reply by mcanouil** [reply]
Indeed, I posted after the other marked answer ;)
